universal-login-react/src/ui/BackupCodes/BackupCodes.tsx

```diff
@@ -1,58 +1,68 @@
 import React, {useState} from 'react';
+import {join} from 'path';
+import {Switch, Route, useHistory} from 'react-router';
 import {DeployedWallet} from '@universal-login/sdk';
-import BackupCodesLoader from './BackupCodesLoader';
 import BackupCodesView from './BackupCodesView';
 import './../styles/backup.sass';
 import './../styles/backupDefault.sass';
 import {BackupCodesInitial} from './BackupCodesInitial';
 import {BackupCodesWrapper} from './BackupCodesWrapper';
 import {ErrorMessage} from '../commons/ErrorMessage';
+import {WaitingForTransaction} from '../commons/WaitingForTransaction';
 
 export interface BackupProps {
   deployedWallet: DeployedWallet;
+  basePath?: string;
   className?: string;
 }
 
-type BackupState = 'Initial' | 'Loading' | 'Generated' | 'Failure';
-
-export const BackupCodes = ({deployedWallet, className}: BackupProps) => {
+export const BackupCodes = ({deployedWallet, basePath = '', className}: BackupProps) => {
   const [backupCodes, setBackupCodes] = useState<string[]>([]);
-  const [state, setState] = useState<BackupState>('Initial');
+  const history = useHistory();
+  const relayerConfig = deployedWallet.sdk.getRelayerConfig();
 
   const generateBackupCodes = async () => {
-    setState('Loading');
     try {
-      const {waitToBeSuccess} = await deployedWallet.generateBackupCodes();
+      history.push(join(basePath, 'waitingForBackupCodes'));
+      const {waitToBeSuccess, waitForTransactionHash} = await deployedWallet.generateBackupCodes();
+      const {transactionHash} = await waitForTransactionHash();
+      history.push(join(basePath, 'waitingForBackupCodes'), {transactionHash});
```

Comment:
```suggestion
      history.replace(join(basePath, 'waitingForBackupCodes'), {transactionHash});
```

Reply:
I agree, there are more places to change this. Let's create the next PR for this